### PR TITLE
Add support for Django 4

### PR DESCRIPTION
**Pull request by rafikdraoui**
There is no change to the application code itself: the only required change is in the test application.

### CODE REVIEW

**Comment by jonprindiville** [comment]
Huh... as far as "make it run tests for Django 4" seems like it did that without anything failing ✅

---

I wish that I had looked a little closer on [that previous PR](https://github.com/gadventures/django-gapi-hooked/pull/25), though, cause now when I do I get to wondering if we've misunderstood something in the tox setup cause it kind of feels like all the environments are not running 😕 

Its definitely possible that I am misunderstanding, but the thing that makes me feel that way is that I only see one venv getting set up in each of our Circle py37/py38/py39, e.g. [py310 got Django 4, but nothing else from what I can see](https://app.circleci.com/pipelines/github/gadventures/django-gapi-hooked/27/workflows/ec7a911e-0b94-4c0f-aae9-008614330f6c/jobs/102?invite=true#step-103-9)

A funny thing is that when I `tox -l` locally I do get all of the expected environments. Also, it is possible for me to `tox -e py37-django111,py37-django22,py37-django32,py37-django40`, for example, and do multiple environments in a single invocation.

**Comment by jonprindiville** [comment]
After some Googling and reading... maybe that's just not how tox works?

[Here's a thread about "how can I select envs based on a wildcard?"](https://github.com/tox-dev/tox/issues/318) and the conclusion seems to be _oh if you want to select by a single factor, maybe try this tox plugin?_

**Comment by jonprindiville** [comment]
Diving into tox docs also reminds me that [_`setup.py test` is deprecated and will be removed in a future version._](https://tox.wiki/en/latest/example/basic.html#integration-with-setup-py-test-command) but that's more "would have been good to talk about that on the previous PR" kind of stuff

File: tox.ini [hunk 1]
```diff
@@ -1,9 +1,10 @@
 [tox]
-envlist = py{37,38,39}-django{111,22,32},py310-django{22,32}
+envlist = py{37,38,39}-django{111,22,32,40},py310-django{22,32,40}
```
Comment (jonprindiville) [inline tox.ini:2]: Ah, when I manually selected all of the py37 environments to run locally (`tox -e py37-django111,py37-django22,py37-django32,py37-django40`) I learned that Django 4.x will not install there -- [only Python 3.8, 3.9, 3.10 supported](https://docs.djangoproject.com/en/4.0/releases/4.0/#python-compatibility)

Maybe it's this? 👇 

```suggestion
envlist = py37-django{111,22,32}, py{38,39}-django{111,22,32,40}, py310-django{22,32,40}
```

**Comment by rafikdraoui** [comment]
Thanks for catching that! It looks like all environments are tested now: https://app.circleci.com/pipelines/github/gadventures/django-gapi-hooked/29/workflows/5ee46167-ce6a-49de-b1d6-aaba94f7cd86

**Review by jonprindiville — APPROVED** [review]
🔢 🎉

**Pull request merged**